src/syft/core/node/device/device.py

```diff
@@ -40,10 +40,12 @@ def __init__(
         device: SpecificLocation = SpecificLocation(),
         vm: Optional[Location] = None,
         device_type: DeviceType = unknown_device,
-        vms: Dict[UID, VirtualMachine] = {},
+        vms: Dict[UID, VirtualMachine] = None,
```

Comment:
This used to be an Optional so i think it has just been wrongly changed, but perhaps it can be refactored by removing the default below...

Reply:
We need to be careful here since the concrete child classes actually pass through None:
```python
def __init__(
        self,
        *,  # Trasterisk
        name: Optional[str] = None,
        network: Optional[Location] = None,
        domain: Optional[Location] = None,
        device: Optional[Location] = None,
        vm: SpecificLocation = SpecificLocation(),
        signing_key: Optional[SigningKey] = None,
        verify_key: Optional[VerifyKey] = None,
    ):
        super().__init__(
            name=name,
            network=network,
            domain=domain,
            device=device,
            vm=vm,
            signing_key=signing_key,
            verify_key=verify_key,
        )
```

So I think these are actually correct.

```python
# must be supplied and defaults to {}
vms: [Dict[UID, VirtualMachine]] = {}
```
If we call this parent constructor with a `None`, we need to then detect that down stream and overwrite it since an explicit None here from the subclass won't use the default...